io/disk/dlpar_vscsi.py

```diff
@@ -34,7 +35,8 @@ def setUp(self):
         Gather necessary test inputs.
         '''
         device = self.params.get('disk', default=None)
-        self.disk = disk.get_absolute_disk_path(device)
+        dev_path = disk.get_absolute_disk_path(device)
+        self.disk = os.path.basename(os.path.realpath(dev_path))
```

Comment:
@abdhaleegit what are we trying to achieve here,
```

>>> disk.get_absolute_disk_path('/dev/sda')
'/dev/sda'
>>> 
>>> disk.get_absolute_disk_path('sda')
'/dev/sda'
>>> 
>>> os.path.basename(os.path.realpath('/dev/sda'))
'sda'
>>> 
```

```
# lsscsi 
[0:0:1:0]    disk    AIX      VDASD            0001  /dev/sda 
[2:0:1:0]    disk    AIX      VDASD            0001  /dev/sdb 

====>>> above two disks are vscsi disks

[root@sys avocado-fvt-wrapper]# lsblk -l sda
lsblk: sda: not a block device
[root@sys avocado-fvt-wrapper]# 
[root@sys avocado-fvt-wrapper]# lsblk -l /dev/sda
NAME                          MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sda                             8:0    0   70G  0 disk 
sda1                            8:1    0    4M  0 part 
sda2                            8:2    0    1G  0 part /boot
sda3                            8:3    0   69G  0 part 
rhel_sys-root 253:0    0 41.7G  0 lvm  /
rhel_sys-swap 253:1    0    7G  0 lvm  [SWAP]
rhel_sys-home 253:2    0 20.3G  0 lvm  /home
[root@sys avocado-fvt-wrapper]#
```

I feel the older way itself working fine.
can you please re-check on this

Reply:
@Naresh-ibm my input here is /dev/disk/by-id/scsi-3224232323.2   so the test were failing

Reply:
all our input to test are device by id for persistent device name